vendor/engines/secure_rooms/app/controllers/secure_rooms_api/scans_controller.rb

```diff
@@ -10,19 +10,38 @@ class SecureRoomsApi::ScansController < ApplicationController
   before_action :load_models
 
   def scan
-    response_json = {
-      response: "deny",
-      reason: "I only know how to deny right now.",
-    }
-
-    render json: response_json, status: :forbidden
+    user_accounts = @user.accounts_for_product(@card_reader.secure_room)
```

Comment:
I think we're already starting to see the need for some kind of service object to keep this controller fresh and clean. Dunno that you have to do it in this PR, but I wouldn't wait too long past it, since this code is only going to get hairier (and it's gonna be nice to test all its various permutations outside the context of a controller environment).